tiledb/oxidize/CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+# assumes cargo is found via `Modules/FindCargo.cmake`
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(CARGO_PROFILE "dev")
+    set(CARGO_TARGET "debug")
+else ()
+    set(CARGO_PROFILE "release")
+    set(CARGO_TARGET "release")
+endif ()
+
+set(CARGO_MANIFEST_PATH "${CMAKE_CURRENT_SOURCE_DIR}/Cargo.toml")
+
+set(CARGO_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/target")
+set(CARGO_LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/target/${CARGO_TARGET}")
+
+set(CARGO_VERBOSITY "-q")
+
+
+# Declare platform-specific parameters
+if (WIN32)
+    # A Rust deficiency is that it only links with the release CRT.
+    # Fortunately it is possible to work around this by linking the debug runtime library
+    # using CFLAGS and CXXFLAGS.
+    #
+    # https://github.com/rust-lang/rust/issues/39016#issuecomment-853964918
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(TILEDB_OXIDIZE_CFLAGS "-MDd")
+    endif ()
+
+    # It is also necessary for some reason to specify the system libraries we need
+    # to link with.  This should probably be obtained more scientfically
+    # (see `corrosion`'s use of `cargo rustc` for example) but this does the job.
+    #
+    # https://github.com/corrosion-rs/corrosion/blob/715c235daef4b8ee67278f12256334ad3dd4c4ae/cmake/FindRust.cmake#L155
+    set(TILEDB_OXIDIZE_LINK_LIBS "kernel32.lib;advapi32.lib;ws2_32.lib;ntdll.lib;userenv.lib")
+endif ()
+
+
+# Declares targets for building a Rust crate.
+
+# Inputs:
+#   NAME crate_name
+#   SOURCES source1 source2 ...
+#
+# Outputs:
+#   ${crate_name}_rs object library target
+#   Building this target produces the object library and also the `#[cxx::bridge]` files,
+#   which can be included in C++ source files via `#include "tiledb/oxidize/${crate_name}.h".
+function(add_crate)
```

Comment:
Should this function be moved to a separate file in `cmake/`?

Reply:
Done 05720f9532